src/Kiota.Builder/Writers/Go/CodeEnumWriter.cs

```diff
@@ -44,29 +44,35 @@ public override void WriteCodeElement(CodeEnum codeElement, LanguageWriter write
                         string.Empty,
                         $"func (i {typeName}) String() string {{");
         writer.IncreaseIndent();
+        writer.WriteLine($"var values []string");
```

Comment:
This generation of multi-value enum should be conditional on the `IsFlags` property of the CodeEnum and not for all enum values. Unless this is a pattern/experience acceptable in Golang.
Currently looking into having this property propagated to the openApi metadata so that it's set properly at https://github.com/microsoft/kiota/blob/bf0f64e5e1242ea0fd6b287c225b565b1f526b8f/src/Kiota.Builder/KiotaBuilder.cs#L1702 asap.